homeassistant/components/homekit_controller/climate.py

```diff
@@ -166,14 +164,6 @@ def _update_relative_humidity_target(self, value):
     @property
     def state(self):
         """Return the current state."""
-        # If the device reports its operating mode as off, it sometimes doesn't
-        # report a new state.
-        if self._current_mode == HVAC_MODE_OFF:
-            return HVAC_MODE_OFF
-
-        if self._state == HVAC_MODE_OFF and \
-                self._current_mode != HVAC_MODE_OFF:
-            return HVAC_MODE_OFF
```

Comment:
Remove the state overwrite completely because that is something that the base entity object do

Reply:
Done